### ISSUE TITLE
YOLOv8 Architecture Design Verification and Citation

### ISSUE THREAD

**Issue by adnan10tc39** [ultralytics/ultralytics#7964]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I designed the YOLOv8 architecture where I mainly took help from the Ultralytics YOLOv8 code, and I also watched Dr. Priyanto Hidayatullah's tutorials on YOLOv8 which were quite helpful. Now, the problem is whether this architecture is correct or not. Secondly, if I use it in my work, do I need to only cite Ultralytics, or do I need to include any other citations or required permissions?
Architecture Diagrams are given below @glenn-jocher 
![yolov8_architecture](https://github.com/ultralytics/ultralytics/assets/81921747/6fa2e3e4-1a71-49d1-a3c3-d12430cbeed4)

![yolov8_architecture_blocks](https://github.com/ultralytics/ultralytics/assets/81921747/755d8eee-25b0-446c-8644-59a7be433f5a)

**Comment by adnan10tc39** [comment]
Sorry circle represents (shortcut=True) and square represents the shortcut=false.
![dsjhcj](https://github.com/ultralytics/ultralytics/assets/81921747/25420be2-16f6-4e78-ace7-8a41b55006a4)

**Comment by glenn-jocher** [comment]
@adnan10tc39 hello! It's great to hear that you've designed an architecture based on YOLOv8 and have found the resources helpful. To verify the correctness of your architecture, you should compare it against the official documentation and the YOLOv8 model configuration files in our repo.

Regarding citations, if you've utilized the Ultralytics YOLOv8 codebase or concepts, you should cite the Ultralytics YOLOv8 repository. If Dr. Priyanto Hidayatullah's tutorials have significantly contributed to your work, it would be appropriate to cite those as well, following standard academic practices.

For permissions, the YOLOv8 models are available under the AGPL-3.0 License, which allows for use and modification as long as the same license is maintained. If you require a more flexible license for commercial purposes, consider obtaining an Enterprise License.

For further details on licensing and citations, please refer to our documentation at https://docs.ultralytics.com. Good luck with your work! 🚀

**Comment by Burhan-Q** [comment]
FYI, the model structure always prints at the start of training:

```
                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]
 22        [15, 18, 21]  1    897664  ultralytics.nn.modules.head.Detect           [80, [64, 128, 256]]

Model summary: 225 layers, 3157200 parameters, 3157184 gradients, 8.9 GFLOPs
```
